### DISCUSSION TITLE
Looking for git workflow idea for short, infrequent sprints

### DISCUSSION THREAD

**Discussion by vhe1** [community/community#21874] [New to GitHub]
<p>Hi!</p>
<p>We just got a legacy project (<a href="http://asp.net" rel="nofollow noopener">asp.net</a> mvc app of, say - questionable - quality), are four developers plus one product owner (we do scrum) and we are supposed to work on it for one or two days every month, possibly even less, i.e. a couple of days every few months.</p>
<p>The current flow is that we branch features off develop and have no other branches. The workflow is that</p>
<ol>
<li>We develop each feature on a branch and test.</li>
<li>After or near the end of the sprint, another tester tests each feature branch.</li>
<li>After that, the stuff that works gets merged to develop.</li>
<li>Develop gets tested again.</li>
<li>Things get deployed.</li>
</ol>
<p>The problem I have is that, due to the short duration of the sprint, we can’t really merge stuff back to develop for other feature branches to be updated with finished features. So, bugs introduced by merging are discovered late, if at all.</p>
<p>Also, we have three kinds of tasks:</p>
<ol>
<li>Small, isolated tasks (change the text on a view).</li>
<li>Tasks that may break stuff (adding a field to a database).</li>
<li>Refactorings which, by their nature, affect lots of files.</li>
</ol>
<p>Now, is the flow we have ok for these tasks? Should we use different ways of working with the different task categories?</p>
<p>Would the full git flow be an advantage? Or does anyone have an even better idea?</p>
<p>Any help would be greatly appreciated.</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>This isn’t really a Git question, but more of a general software development philosophy question :grinning:</p>
<p>It sounds to me like the lion’s share of the testing occurs in step 2 of your flow and step 4 is more of a “sanity check”. If you’re not catching enough bugs pre-merge, then your process should focus its testing effort on testing post-merge.</p>
<p>Your three classes of changes are pretty much standard. Every software project has some of each of those changes.</p>
<p>I’m not sure what you mean by “the full git-flow”. It sounds like you’re doing that already. Here’s <a href="https://guides.github.com/introduction/flow/" rel="nofollow noopener">the process that we recommend for use on GitHub</a>. We recommend this process for changes of all sizes. If some changes are particularly large or long-running, then we tend to merge from the default branch (<code>develop</code> in your case) into the PR branch at regular intervals along the way. We do this to ensure that we’re testing the right thing all the time.</p>
<p>Do you have automated tests helping with the testing workload?</p>

**Comment by vhe1** [comment]
<p>I think that has cleared it up for me beautifully, including that link to the flow you use.</p>
<p>Thanks a lot!</p>
<p>(Yes, we do have automated tests, which could use some improvement, but <em>that</em> need is clear and the flow is not intended to replace automated tests.)</p>
